Client/Frontend/Browser/MailProviders.swift

```diff
@@ -7,75 +7,115 @@ import Shared
     func newEmailURLFromMetadata(_ metadata: MailToMetadata) -> URL?
 }
 
 extension MailProvider {
-    fileprivate func constructEmailURLString(_ urlFormat: String,
+    fileprivate func constructEmailURLString(_ urlComponents: URLComponents,
                                              metadata: MailToMetadata,
                                              supportedHeaders: [String],
                                              bodyHName: String = "body",
                                              toHName: String = "to",
-                                             toParamFormat: String = "%@=%@") -> String {
+                                             emailFormat: MailProviderEmailFormat = .standard) -> URLComponents {
         var lowercasedHeaders = prepareHeaders(metadata: metadata)
 
         // Evaluate "to" parameter
-        var toParam: String
-        if let toHValue = lowercasedHeaders["to"] {
-            let value = metadata.to.isEmpty ? toHValue : [metadata.to, toHValue].joined(separator: "%2C%20")
+        var toQueryItem: URLQueryItem
+        if let toHeaderValue = lowercasedHeaders["to"] {
+            let value = metadata.to.isEmpty ? toHeaderValue : [metadata.to, toHeaderValue].joined(separator: ",")
             lowercasedHeaders.removeValue(forKey: "to")
-            toParam = String(format: toParamFormat, toHName, value)
+            toQueryItem = URLQueryItem(name: toHName, value: value)
         } else {
-            toParam = String(format: toParamFormat, toHName, metadata.to)
+            toQueryItem = URLQueryItem(name: toHName, value: metadata.to)
         }
 
         // Create string of additional parameters
-        let stringParams = prepareParams(headers: lowercasedHeaders,
-                                         supportedHeaders: supportedHeaders,
-                                         bodyHName: bodyHName)
+        let additionalQueryItems = prepareParams(headers: lowercasedHeaders,
+                                                 supportedHeaders: supportedHeaders,
+                                                 bodyName: bodyHName)
 
         // Create url based on scheme
-        var finalURLString = String(format: urlFormat, toParam, (stringParams.isEmpty ? "" : stringParams))
+        var urlComponents = urlComponents
+        var queryItems: [URLQueryItem] = [toQueryItem]
+        queryItems.append(contentsOf: additionalQueryItems)
+        urlComponents.queryItems = queryItems.isEmpty ? nil : queryItems
+
+        if emailFormat == .userAndPassword {
+            urlComponents = addUserAndPasswordComponents(urlComponents, toQueryItem: toQueryItem)
+        }
+
+        return urlComponents
+    }
+
+    // Used to build Protonmail URL correctly (format: "protonmail://mailto:email@test.com")
+    // We use the user, password and host component parts for this URL format.
+    private func addUserAndPasswordComponents(_ components: URLComponents,
```

Comment:
note: Is this only for proton mail then could we clean this up for proton mail

I like the comments though!! Keep them as they are helpful to understand